### DISCUSSION TITLE
x-bind:style converts CSS variables to lowercase

### DISCUSSION THREAD

**Discussion by justrau** [alpinejs/alpine#2577] [7. Bugs (archived)]
When using x-bind:style to set values to CSS variables, names of these variables are converted to lowercase. This is an issue, because CSS variables are case-sensitive (https://developer.mozilla.org/en-US/docs/Web/CSS/Using_CSS_custom_properties, "Note: Custom property names are case sensitive — --my-color will be treated as a separate custom property to --My-color.").

I think that this conversion is done at https://github.com/alpinejs/alpine/blob/6de8c65e781dc8ab4f8953d15a88049cf1fda8c1/packages/alpinejs/src/utils/styles.js#L19 and is a by-product of other functionality (converting camelCase to kebab-case).

**Issue reproduction**
https://jsfiddle.net/justasraudonius/vmjx1s7d/12/
`--Polaris-RangeSlider-current` and `--Polaris-RangeSlider-progress` are needed to properly style the range slider as its value changes, but after binding they become `--polaris-range-slider-current` and `--polaris-range-slider-progress` respectively.

**Proposed solution**
Add a check in `setStylesFromObject(el, value)` to not apply `kebabCase(key)` when `key` starts with `--`, which is the standard notation for declaring CSS variables.

I could try to PR a fix.

**Comment by SimoTod** [comment]
Hi @justRau 

good shout, I implemented that functionality and the reason for using kebab-case was that standard css properties in js are camelCase strings (see https://github.com/alpinejs/alpine/pull/1920 for more details) and it's actually used in that way by x-transition.

You are correct though, the conversion is probably too aggressive and it shouldn't happen if the initial string is a custom CSS property. It shouldn't be too hard to send PR to apply kebab case only if the property doesn't start with `--` (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/startsWith).

Probably worth adding a test as well.

**Reply by justrau** [reply]
@SimoTod thanks for a quick and detailed answer!

I've implemented the fix together with a test in PR #2581.
